Description: Remove libusb.h include from exported header
 Instead of declaring freenect_usb_context as libusb_context, declare it as
 void. In the first place, it was not necessary since it is always passed as
 pointer to the API and was already void on Win32. Secondly this fixes the
 problem of discrepancy between the include path of libusb on freebsd and
 the other architectures.
Origin: vendor
Forwarded: yes
Applied-Upstream: commit:3d735a89dfc380e8ff346a6815be2b54e4fafdfb
Author: Nicolas Bourdaud <nicolas.bourdaud@gmail.com>
Last-Update: 2012-03-04

--- a/include/libfreenect.h
+++ b/include/libfreenect.h
@@ -29,6 +29,13 @@
 
 #include <stdint.h>
 
+/* We need struct timeval */
+#ifdef _WIN32
+#include <winsock.h>
+#else
+#include <sys/time.h>
+#endif
+
 #ifdef __cplusplus
 extern "C" {
 #endif
@@ -156,13 +163,7 @@
 typedef struct _freenect_device freenect_device; /**< Holds device information. */
 
 // usb backend specific section
-#ifdef _WIN32
-  /* frees Windows users of the burden of specifying the path to <libusb-1.0/libusb.h> */
-  typedef void freenect_usb_context;
-#else
-  #include <libusb-1.0/libusb.h>
-  typedef libusb_context freenect_usb_context; /**< Holds libusb-1.0 specific information */
-#endif
+typedef void freenect_usb_context; /**< Holds libusb-1.0 context */
 //
 
 /// If Win32, export all functions for DLL usage
--- a/src/libfreenect.pc.in
+++ b/src/libfreenect.pc.in
@@ -5,7 +5,7 @@
 	
 Name: @CMAKE_PROJECT_NAME@
 Description: Interface to the Microsoft Kinect sensor device.
-Requires: libusb-1.0
+Requires.private: libusb-1.0
 Version: @PROJECT_APIVER@
 Libs: -L${libdir} -lfreenect
 Cflags: -I${includedir}
